File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/participant/remote/RemoteParticipantJoinedHandler.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.android.communication.ui.participant.remote;

import com.azure.android.communication.common.CommunicationIdentifier;

public interface RemoteParticipantJoinedHandler {
    void handle(CommunicationIdentifier communicationIdentifier, RemoteParticipantManager remoteParticipantManager);
}

File: azure-communication-ui/azure-communication-ui/src/main/java/com/azure/android/communication/ui/persona/RemoteParticipantJoinedHandler.kt
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.android.communication.ui.persona

import com.azure.android.communication.ui.configuration.CallCompositeConfiguration
import com.azure.android.communication.ui.redux.Store
import com.azure.android.communication.ui.redux.state.ReduxState
import kotlinx.coroutines.flow.collect

internal class RemoteParticipantJoinedHandler(
    private val configuration: CallCompositeConfiguration,
    private val store: Store<ReduxState>,
) {

    suspend fun start() {
        store.getStateFlow().collect {
            onStateChanged(it)
        }
    }

    private fun onStateChanged(state: ReduxState) {
    }
}
